### PR TITLE
New Providers [Vietnamese][Ukrainian][Russian]

### LINKED ISSUE #1282: issues with oploverz and nontonanimeid
### Source information

Oploverz and NontonAnimeID

### Source link

https://65.108.132.145 and https://75.119.159.228

### Steps to reproduce

there are some anime with oploverz and nontonanimeid providers that cannot be opened in the application but can be opened on the website.
Screenshot_20220712-052605_CloudStream_Beta
Screenshot_20220712-052614_CloudStream_Beta

### CloudStream version

2.10.30-PRE

### Android version

Android 9

### Other details

_No response_

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I have updated the app to pre-release version **[Latest](https://github.com/LagradOst/CloudStream-3/releases)**.
- [X] If related to a provider, I have checked the site and it works, but not the app.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by hexated**
### Provider

- [Phimmoichill](https://phimmoichill.net) - **Vietnamese**
- [Uakino](https://uakino.club) - **Ukrainian**
- [HDrezka](https://rezka.ag) - **Russian**

### Others
- enable [indexsubtitle ](https://indexsubtitle.com/)
- fix kuramanime
- add extractor
- fixes #1282

### CODE REVIEW

**Comment by Blatzar** [comment]
[I added segmented stream support](https://github.com/LagradOst/CloudStream-3/commit/2ca8f6f870ae7703cd63b13c714143b030a906ee)
Try using `ExtractorLinkPlayList` instead of `ExtractorLink`
I don't think seeking works and I am not sure how fixable that is, but will investigate more tomorrow :pensive:

**Comment by hexated** [comment]
> [I added segmented stream support](https://github.com/LagradOst/CloudStream-3/commit/2ca8f6f870ae7703cd63b13c714143b030a906ee) Try using `ExtractorLinkPlayList` instead of `ExtractorLink` I don't think seeking works and I am not sure how fixable that is, but will investigate more tomorrow 😔

yap.. still doesn't work

**Comment by Blatzar** [comment]
> > [I added segmented stream support](https://github.com/LagradOst/CloudStream-3/commit/2ca8f6f870ae7703cd63b13c714143b030a906ee) Try using `ExtractorLinkPlayList` instead of `ExtractorLink` I don't think seeking works and I am not sure how fixable that is, but will investigate more tomorrow pensive
> 
> yap.. still doesn't work

Looking at the code I think you misunderstood durationUs, it's microseconds. I think that's why it fails, the total stream doesn't amount to more than a fraction of a second

**Comment by hexated** [comment]
> > > [I added segmented stream support](https://github.com/LagradOst/CloudStream-3/commit/2ca8f6f870ae7703cd63b13c714143b030a906ee) Try using `ExtractorLinkPlayList` instead of `ExtractorLink` I don't think seeking works and I am not sure how fixable that is, but will investigate more tomorrow pensive
> > 
> > 
> > yap.. still doesn't work
> 
> Looking at the code I think you misunderstood durationUs, it's microseconds. I think that's why it fails, the total stream doesn't amount to more than a fraction of a second

is that means like this..?

```kotlin
app.get("https://so-trym.topphimmoi.org/hlspm/$key", referer = "$mainUrl/")
            .parsedSafe<ResponseM3u>()?.main.let {
            val playList = it?.segments?.map { segment ->
                val duration = (segment.du.toFloat() * 1_000_000).toLong()
                PlayListItem(
                    segment.link,
                    duration
                )
            }
```

**Comment by Blatzar** [comment]
Yes :+1:

**Comment by hexated** [comment]
you can test now

File: app/src/main/java/com/lagradost/cloudstream3/movieproviders/PhimmoichillProvider.kt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+            }
+
+            callback.invoke(
+                ExtractorLinkPlayList(
+                    "PMFAST",
+                    "PMFAST",
+                    playList ?: return@let,
+                    referer = "$mainUrl/",
+                    quality = Qualities.Unknown.value,
+                    isM3u8 = true
```
Comment (Blatzar) [inline app/src/main/java/com/lagradost/cloudstream3/movieproviders/PhimmoichillProvider.kt:176]: This is causing the issue, you don't have a playlist of m3u8 links, your playlist items are videos. Set to false

**Comment by hexated** [comment]
**it works now, but it left one problem. duration of the video changes to the duration of the PlayListItem** 

<image src="https://user-images.githubusercontent.com/37908684/178801757-bc657bd2-42cf-4e95-b6c3-882834aad30c.png" width="600px" height="360px"></image>

**Comment by hexated** [comment]
you can merge it now. PlayListItem only used for Phimmoichill main server, backup server works well..

**Comment by zeref-dragneel** [comment]
Can you help me add a Vietnamese site for anime? If you agree, I will list Vietnamese anime websites for you to check.

**Comment by zeref-dragneel** [comment]
Please choose one of the sites I recommend that you can add. Thank you.

**Comment by hexated** [comment]
put your sites in here #24 and add one or two your best site. i will check if i can add one

**Comment by zeref-dragneel** [comment]
I have listed a list of Vietnamese anime websites. Hopefully you can choose one or two sites you can add to the app.

**Comment by zeref-dragneel** [comment]
Phimmoichill.net not working. 😃

**Pull request merged**